### PR TITLE
Tag feature

### PR DESCRIPTION
**Pull request by chrisliang12**
1. Implemented tag feature
2. rebuild on top of the modify branch, should not have conflicts
3. please check if there are any bugs in tag feature

### CODE REVIEW

File: src/scripts/home.js [hunk 1]
```diff
@@ -304,22 +266,131 @@ function delete_profile(profile) {
...
- *    such a tag. if it has, add this profile object to a
- *    temp list.
- * 2. remove all card components in grid
- * 3. call add_profiles_to_doc(profiles) to display the
- *    profile we just added to the temp list
- *
- * @param {String} tag an tag
- */
-function sort_by_tag(tag) {}
+function search_tag(tag) {
```
Comment (benjxia) [inline src/scripts/home.js:278]: documentation pls

File: src/scripts/home.js [hunk 2]
```diff
@@ -304,22 +266,131 @@ function delete_profile(profile) {
...
+      !match_list.includes(profile.tag) &&
+      profile.tag.includes(tag)
+    ) {
+      match_list.push(profile.tag);
+    }
+  });
+  return match_list;
+}
+
+function create_tag_btn() {
```
Comment (benjxia) [inline src/scripts/home.js:292]: documentation pls

File: src/scripts/home.js [hunk 3]
```diff
@@ -304,22 +266,131 @@ function delete_profile(profile) {
...
+      is_find = true;
+    }
+  }
+  if (!is_find) {
+    console.log('zzz')
+    all_btn.click();
+  }
+}
+
+function handle_tag_btn_click(curr_tag_btn, profile) {
```
Comment (benjxia) [inline src/scripts/home.js:352]: documentation pls

File: src/scripts/setup.js [hunk 4]
```diff
@@ -57,7 +68,35 @@ function setup_delete() {
   });
 }
 
-export {
-  setup_modify,
-  setup_delete
-};
+export function setup_tag_recommend(search_tag) {
```
Comment (benjxia) [inline src/scripts/setup.js:71]: documentation

**Review by benjxia — COMMENTED** [review]
Are profiles limited to one tag for this version?

**Comment by benjxia** [comment]
ill write the jsdoc stuff so i can learn this stuff better

**Comment by benjxia** [comment]
confirm cancel modal always shows up even when there are no changes

File: src/scripts/setup.js [hunk 5]
```diff
@@ -57,7 +68,35 @@ function setup_delete() {
...
+  });
+
+  // set up listener for info_modal
+  const info_modal_tag = document.querySelector('#info-modal-input-tag');
+  info_modal_tag.addEventListener('input', () => {
+    handle_tag_input_change(search_tag, info_modal_tag, 'info');
+  });
+}
+
+function handle_tag_input_change(search_tag, tag_input, type) {
```
Comment (benjxia) [inline src/scripts/setup.js:85]: why was search tag passed through 3 functions instead of just exporting from `home.js`?

Comment (benjxia) [inline src/scripts/setup.js:85]: why is search_tag() passed through 3 functions instead of directly exporting from home.js?

**Comment by chrisliang12** [comment]
> ill write the jsdoc stuff so i can learn this stuff better

I'll add them later (tomorrow night or Saturday) so don't merge this pr for now.

File: src/scripts/home.js (hunk 2 above)
Comment (benjxia) [reply src/scripts/home.js:292]: this one im having trouble understanding rn, my brain is too fried rn since its late

File: src/scripts/home.js [hunk 6]
```diff
@@ -304,22 +266,145 @@ function delete_profile(profile) {
...
+    profile_list.forEach((profile_temp) => {
+      if (profile === 'all' || (profile_temp.tag && profile_temp.tag === profile.tag)) {
+        profile_list_tag.push(profile_temp);
+      }
+    });
+    display_selected_profile(profile_list_tag);
+  }
+}
+
+function display_selected_profile(profiles) {
```
Comment (benjxia) [inline src/scripts/home.js:385]: doc

**Comment by chrisliang12** [comment]
If there are no bugs, it is ready to merge

**Review by benjxia — APPROVED** [review]
couldnt break it, guessing multi filter will be part 2, just convert the profile.tag from string to array of strings later

**Comment by benjxia** [comment]
this branch isnt completely backwards compatible with old profiles (confirm cancel modal will always show up since old profiles have null tag). don't worry about fixing it though just telling other people who test this branch

**Comment by benjxia** [comment]
dont merge yet, im partially done with a multi-filter prototype

**Comment by chrisliang12** [comment]
> dont merge yet, im partially done with a multi-filter prototype

For that part, I think you can do it on another branch. It is not a part of phase 1.

**Comment by benjxia** [comment]
yes it's on a different branch, i want to try to PR into this branch once it's done

**Comment by chrisliang12** [comment]
> yes it's on a different branch, i want to try to PR into this branch once it's done

OK. That is fine then. I can also look at them later today.

**Review by kjgr — APPROVED** [review]

**Review by karthikv6 — APPROVED** [review]

**Review by Anishkulkarni9001 — APPROVED** [review]

**Review by renawuq — APPROVED** [review]

**Pull request merged**